src/azure-cli/azure/cli/command_modules/cdn/_params.py

```diff
@@ -154,3 +140,55 @@ def load_arguments(self, _):
     # Origin #
     with self.argument_context('cdn origin') as c:
         c.argument('origin_name', name_arg_type, id_part='name')
+
+    # WAF #
+    with self.argument_context('cdn waf') as c:
+        c.argument('rule_set_type', help='The type of the rule set.')
+        c.argument('rule_set_version', help='The version of the rule set.')
+
+    with self.argument_context('cdn waf managed-rule-set rule-group') as c:
+        c.argument('name', name_arg_type, id_part='name', help='The name of the rule group.')
+
+    with self.argument_context('cdn waf policy set') as c:
+        c.argument('disabled', arg_type=get_three_state_flag())
+        c.argument('block_response_status_code', type=int)
+        c.argument('name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+    with self.argument_context('cdn waf policy show') as c:
+        c.argument('policy_name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+    with self.argument_context('cdn waf policy delete') as c:
+        c.argument('policy_name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+
+    with self.argument_context('cdn waf policy managed-rule-set') as c:
+        c.argument('policy_name', help='Name of the CDN WAF policy.')
+        c.argument('rule_set_type', help='Type name of the managed rule set.')
+        c.argument('rule_set_version', help='Version the managed rule set.')
+    with self.argument_context('cdn waf policy managed-rule-set add') as c:
+        c.argument('enabled', arg_type=get_three_state_flag())
+
+    with self.argument_context('cdn waf policy managed-rule-set rule-group-override') as c:
```

Comment:
What do you mean for rule-group-override?

Reply:
In our service, you can use a rule group override to override the default rules applied by adding a managed rule set to a policy. I considered shortening this to `cdn waf policy managed-rule-set rule-group`, but I wanted to make it clear that overriding the defaults of one policy doesn't modify the managed rule set which may be applied to other policies. This also follows the naming convention used by our service's swagger API.